language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/ActiveContracts.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.javaapi.data;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+public final class ActiveContracts<Ct> implements WorkflowEvent {
+
+  private final String offset;
+
+  private final List<Ct> activeContracts;
+
+  private final String workflowId;
+
+  public ActiveContracts(
+      @NonNull String offset, @NonNull List<Ct> activeContracts, String workflowId) {
+    this.offset = offset;
+    this.activeContracts = activeContracts;
+    this.workflowId = workflowId;
+  }
+
+  @NonNull
+  public Optional<String> getOffset() {
+    // Empty string indicates that the field is not present in the protobuf.
+    return Optional.of(offset).filter(off -> !offset.equals(""));
+  }
+
+  @NonNull
+  public List<@NonNull Ct> getContracts() {
+    return activeContracts;
+  }
+
+  @NonNull
+  public String getWorkflowId() {
+    return workflowId;
+  }
```

Comment:
In general, we should define public final properties rather than private properties with getters.

Reply:
You can consider this for a follow-up PR though, e.g. the quickstart-java one.

Reply:
For the record, we do not care that Intellij thinks you shouldn't have `Optional` fields.

Reply:
Sure will do it in the quickstart-java issue: #15191